pkg/console/operator/health.go

```diff
@@ -0,0 +1,74 @@
+package operator
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	deploymentv1 "k8s.io/api/apps/v1"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"github.com/openshift/console-operator/pkg/console/status"
+	deploymentsub "github.com/openshift/console-operator/pkg/console/subresource/deployment"
+	routesub "github.com/openshift/console-operator/pkg/console/subresource/route"
+)
+
+func (co *consoleOperator) CheckRouteHealth(opConfig *operatorv1.Console, rt *routev1.Route) {
+	status.HandleDegraded(func() (conf *operatorv1.Console, prefix string, reason string, err error) {
+		prefix = "RouteHealth"
+		client := http.Client{Timeout: time.Duration(5 * time.Second)}
```

Comment:
You will need the router CA to trust the router certs, get it at the CM `openshift-config-managed/router-ca`, grab the contents of the `ca-bundle.crt` key. Use them in your client's `Transport.TLSClientTransport.RootCAs`

Reply:
That's going to be the first step, you will need to also consider the operator being run in an environment with an MitM proxy

Reply:
Got this, see update & left you a comment regarding `MitM`.